security/cmd/flexvolume/driver/driver.go

```diff
@@ -249,12 +287,120 @@ func failure(caller, inp, msg string) error {
 }
 
 func logToSys(caller, inp, opts string) {
-	if logWrt == nil {
+	if logWriter == nil {
 		return
 	}
 
-	op := caller + "|"
-	op = op + inp + "|"
-	op = op + opts
-	_ = logWrt.Warning(op)
+	opt := strings.Join([]string{caller, inp, opts}, "|")
+	if configuration.LogLevel == logLevelWarn {
+		logWriter.Warning(opt) //nolint: errcheck
+	} else {
+		logWriter.Info(opt) //nolint: errcheck
+	}
+}
+
+func getCredFile(uid string) string {
+	return uid + fv.CredentialFileExtension
+}
+
+// addCredentialFile is used to create a credential file when a workload with the flex-volume volume mounted is created.
+func addCredentialFile(creds *fv.Credential) error {
+	//Make the directory and then write the ninputs as json to it.
+	err := os.MkdirAll(configuration.NodeAgentCredentialsHomeDir, 0755)
+	if err != nil {
+		return err
+	}
+
+	var attrs []byte
+	attrs, err = json.Marshal(creds)
+	if err != nil {
+		return err
+	}
+
+	credsFileTmp := filepath.Join(configuration.NodeAgentManagementHomeDir, getCredFile(creds.UID))
+	err = ioutil.WriteFile(credsFileTmp, attrs, 0644)
+	if err != nil {
+		return err
+	}
+
+	// Move it to the right location now.
+	credsFile := filepath.Join(configuration.NodeAgentCredentialsHomeDir, getCredFile(creds.UID))
+	return os.Rename(credsFileTmp, credsFile)
```

Comment:
Do we need to worry about what happens to the temp file if this Rename() fails? (For example, based on documentation for `os.MkdirAll`, if the directory already exists then `MkdirAll` call above is a noop and the 0755 directory permissions are not applied. So it seems possible at this point that `NodeAgentCredentialsHomeDir` is not writeable by this process.)

Reply:
@justinburke as you know the discussion for this PR is going back to the white board but I also wanted to make sure that I addressed your point here.
If the directory cannot be written to then the driver MUST fail; which i believe it does. `os.Rename` would return an error.
If this PR were to move forward-which i know it isn't at least not in it's current form-i would add a UT for the case you brought up.
:-)